metadoc-js/src/main/scala/metadoc/MetadocApp.scala

```diff
@@ -64,30 +64,22 @@ object MetadocApp extends js.JSApp {
       ScalaLanguageExtensionPoint.id,
       new ScalaDocumentSymbolProvider(index)
     )
+  }
 
-    val options = jsObject[IEditorConstructionOptions]
-    options.readOnly = true
-    val overrides = jsObject[IEditorOverrideServices]
-    val editorService = new MetadocEditorService
-    overrides.textModelResolverService = MetadocTextModelService
-    overrides.editorService = editorService
-    val editor = monaco.editor.Editor.create(app, options, overrides)
-    editor.asInstanceOf[js.Dynamic].getControl = { () =>
-      // NOTE: getControl() is defined on SimpleEditor and is called when changing files.
-      editor
-    }
-    editorService.editor = editor
-    editor.onDidChangeModel((arg1: IModelChangedEvent) => {
-      val path = arg1.newModelUrl.path
-      dom.document.getElementById("title").textContent = path
-      dom.window.location.hash = "#/" + path
-    })
+  def openEditor(editorService: MetadocEditorService, input: IResourceInput): Unit = {
+    for (editor <- editorService.open(input)) {
+      editor.onDidChangeModel((event: IModelChangedEvent) => {
+        val path = event.newModelUrl.path
+        dom.document.getElementById("title").textContent = path
+        dom.window.location.hash = "#/" + path
```

Comment:
We might want to move the `#/` handling into one place someday, no need right now since the hash format i so simple